phase2/outline
# WIP feat/outline-nav-menu

## Description
Still WIP but in a reviewable state
## Adds: `outline-nav` component and sub-components.
## Notes:
Can work with:
1. Pre-formatted data [see default export](packages/outline-nav-menu/src/nav-test-data.ts)
2. JSON:API Menu Items Drupal Module [see example of data model here at jsonData](packages/outline-nav-menu/src/nav-test-data.ts)
3. Can be slots all the way down. [see SlotsTemplate in storybook](packages/outline-storybook/stories/components/outline-nav.stories.ts)

To test with JSON API:
1. Spin up a Drupal demo environment and install the [JSON:API Menu Item module](https://www.drupal.org/project/jsonapi_menu_items)

2. Build your menu in `admin/structure/menu/manage/main`
<img width="810" alt="Cursor_and_Edit_menu_Main_navigation___Phase2_Drupal_Demo" src="https://user-images.githubusercontent.com/11083407/230683978-b29dab0e-37bf-4cf9-b2b7-24dcdbcdce64.png">

3. Spin up an Outline repo and launch storybook.

4. Verify the JSON API endpoint is working at `[docksal evntronment]/jsonapi/menu_items/main` (if you're test menu is the main navigation)

5 Use chrome [Cross Domain CORS]( https://chrome.google.com/webstore/detail/cross-domain-cors/mjhpgnbimicffchbodmgfnemoghjakai/related) extension to allow fetch test
 

6. Paste the endpoint URL into the `menuEndpoint` in the story and refresh.


<img width="745" alt="Cursor_and_Screen_Shot_2023-04-07_at_14_51_09_PM_png" src="https://user-images.githubusercontent.com/11083407/230683503-bc515b15-927c-47cd-bc27-1685f8de82ad.png">


 
<img width="314" alt="Cursor_and_Navigation___Outline_Main_Menu_-_Nav_With_Dynamic_Fetch_URL_⋅_Storybook" src="https://user-images.githubusercontent.com/11083407/230683663-00bdb412-0029-4e99-a565-9f5d6e6e73e0.png">


## Still needs:
1. Styling
2. QOL updates
3. Full accessibility testing.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.


## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Visual Testing
- [ ] Accessibility Testing

## Checklist
